server/src/test/java/org/opensearch/extensions/ExtensionsManagerTests.java

```diff
@@ -528,6 +529,86 @@ public void testHandleExtensionRequest() throws Exception {
         assertEquals("Handler not present for the provided request", exception.getMessage());
     }
 
+    public void testExtensionRequest() throws Exception {
+        ExtensionsManager.RequestType expectedRequestType = ExtensionsManager.RequestType.REQUEST_EXTENSION_DEPENDENCY_INFORMATION;
+
+        // Test ExtensionRequest 2 arg constructor
+        Optional<String> expectedUniqueId = Optional.of("test uniqueid");
+        ExtensionRequest extensionRequest = new ExtensionRequest(expectedRequestType, expectedUniqueId.orElse(null));
+        assertEquals(expectedRequestType, extensionRequest.getRequestType());
+        assertEquals(expectedUniqueId, extensionRequest.getUniqueId());
+
+        // Test ExtensionRequest StreamInput constructor
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            extensionRequest.writeTo(out);
+            out.flush();
+            try (BytesStreamInput in = new BytesStreamInput(BytesReference.toBytes(out.bytes()))) {
+                extensionRequest = new ExtensionRequest(in);
+                assertEquals(expectedRequestType, extensionRequest.getRequestType());
+                assertEquals(expectedUniqueId, extensionRequest.getUniqueId());
+            }
+        }
+
+        // Test ExtensionRequest 1 arg constructor
+        extensionRequest = new ExtensionRequest(expectedRequestType, null);
```

Comment:
Actually use the 1 arg constructor here so you can test the overloaded constructor:
```java
extensionRequest = new ExtensionRequest(expectedRequestType);
```